### PR TITLE
Add YAML front-formatter for all samples need to onboard sample browser.

### PR DESCRIPTION
**Pull request by chenrujun**
## Purpose

Onboarding all samples discussed in https://github.com/Azure-Samples/azure-spring-boot-samples/pull/589 to [sample browser](https://learn.microsoft.com/samples).

Following [this onboarding sample procedure doc](https://review.learn.microsoft.com/en-us/help/contribute/samples/process/onboarding?branch=main#supported-metadata-fields-for-readmemd), we need to add [YAML front-matter](https://review.learn.microsoft.com/en-us/help/contribute/samples/process/onboarding?branch=main#supported-metadata-fields-for-readmemd) for all samples.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## Other Information
1. https://github.com/Azure/azure-sdk-for-java/issues/32148
2. https://review.learn.microsoft.com/en-us/help/contribute/samples/process/onboarding?branch=main#supported-metadata-fields-for-readmemd
3. https://github.com/Azure-Samples/azure-spring-boot-samples/pull/589

### CODE REVIEW

File: aad/spring-cloud-azure-starter-active-directory-b2c/aad-b2c-resource-server/README.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+---
+page_type: sample
+languages:
+- java
+products:
+- azure-active-directory-b2c
+name: Developing REST API application developed by Spring Boot and protected by Azure AD B2C
+description: This sample demonstrates how to develop REST API application developed by Spring Boot and protected by Azure AD B2C.
```
Comment (fangjian0423) [inline aad/spring-cloud-azure-starter-active-directory-b2c/aad-b2c-resource-server/README.md:8]: Azure AD B2C = Azure Active Directory.

I aware description should avoid using abbreviation, all the items should be update. Yesterday's discussion should find it, my mistake.

**Review by fangjian0423 — CHANGES_REQUESTED** [review]

File: aad/spring-cloud-azure-starter-active-directory-b2c/aad-b2c-resource-server/README.md (hunk 1 above)
Comment (chenrujun) [reply aad/spring-cloud-azure-starter-active-directory-b2c/aad-b2c-resource-server/README.md]: Updated.

**Review by fangjian0423 — APPROVED** [review]
looks good

**Comment by chenrujun** [comment]
Now there are 5 results when searching "Spring Boot":
https://learn.microsoft.com/en-us/samples/browse/?terms=Spring%20Boot

> image

I'll share the screenshot after this PR take effect.

**Comment by chenrujun** [comment]
Information sharing:

1. Now these samples can not be searched:
image

2. This json file can be used to check the onboard status: https://github.com/MicrosoftDocs/samples/blob/main/Azure-Samples/azure-spring-boot-samples/.samples-status.json

**Comment by chenrujun** [comment]
Information sharing:
Now samples onboarded:
https://learn.microsoft.com/en-us/samples/browse/?terms=Spring%20Boot
image

**Pull request merged**